strawberry/object_type.py

```diff
@@ -151,6 +148,7 @@ def _process_type(
         _fields=fields,
         is_type_of=is_type_of,
     )
+    cls._type_definition = cls.__strawberry_object__
```

Comment:
could we make this a deprecated property? 😊

Reply:
Since it is used on a class the property would need to use `self`, It is only possible with a metaclass and I wanted to avoid that.

Reply:
Yeah, I looked for it and unfortunately the current `dataclass_transform` specs doesn't allow to define extra attributes that the final type would have. Would be awesome it it allowed though, we would then also be able to mark `__strawberry_object__` as an extra attribute in the type.

Having said that, maybe we could inject a descriptor in `_type_definition` that returns `cls.__strawberry_object__` and raises a deprecation warning when used?

Reply:
That's a good idea

Reply:
I have never implemented a descriptor so I hope I have done it right...